packages/repository/src/__tests__/unit/repositories/legacy-juggler-bridge.unit.ts

```diff
@@ -726,7 +726,9 @@ describe('DefaultCrudRepository', () => {
     });
 
     it(`throws error when execute() not implemented by ds connector`, async () => {
-      delete ds.execute;
+      // TypeScript 4.x fails to compile without cast as `ds.execute` is not
+      // optional
+      delete (ds as {execute?: Function}).execute;
       const repo = new DefaultCrudRepository(Note, ds);
       await expect(repo.execute('query', [])).to.be.rejectedWith(
         'execute() must be implemented by the connector',
```

Comment:
This is interesting. The error message says that `execute` must be implemented by the connector, but we are deleting the method from the datasource. Is this perhaps a bug in the test?